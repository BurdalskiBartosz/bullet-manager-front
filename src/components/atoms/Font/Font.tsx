import { CSSProperties, FC } from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { tTags, tVariants, variants } from 'styles/typography';

type tProps = {
	children: number | string | string[];
	variant?: tVariants;
	component?: tTags;
	style?: CSSProperties;
	[x: string]: any;
};

type tTagProps = {
	$style: string;
	$overideStyle: string;
};
const StyledTag = styled.p<tTagProps>`
	${(props) => props.$style};
	${(props) => props.$overideStyle};
`;

const Font: FC<tProps> = ({ children, variant, component, style, ...rest }) => {
	const { t } = useTranslation();
	let overideStyle;
	let tag: tTags = 'p';
	let variantStyles;
	const textContent =
		typeof children === 'object' || typeof children === 'number'
			? children
			: t(children);

	if (variant) {
		tag = variants[variant].tag;
		variantStyles = variants[variant].styles;
	}

	if (style) {
		overideStyle = Object.entries(style).reduce((acc, [key, value]) => {
			const correctCSSPropertyName = key
				.split(/(?=[A-Z])/)
				.map((el) => el.toLowerCase())
				.join('-');
			acc += `${correctCSSPropertyName}: ${value};`;
			return acc;
		}, '');
	}
	return (
		<StyledTag
			as={component ?? tag}
			$style={variantStyles ?? ''}
			$overideStyle={overideStyle ?? ''}
			{...rest}
		>
			{textContent}
		</StyledTag>
	);
};

export default Font;
