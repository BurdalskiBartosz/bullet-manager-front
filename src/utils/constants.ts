export const baseURL: string = 'http://localhost:3001/api';

export const appRootRoute: string = 'app';
