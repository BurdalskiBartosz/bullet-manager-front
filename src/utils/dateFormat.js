import { format } from 'date-fns';

export const dateFormat = (value) => format(value, 'MM/dd/yyyy');
