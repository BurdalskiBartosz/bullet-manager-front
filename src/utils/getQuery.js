export const getQuery = (object) => {
	if (!object) return '';
	const queryStrings = Object.entries(object).map(([key, value]) => {
		const stringKey = encodeURIComponent(key);
		if (typeof value === 'object')
			return `${stringKey}=${JSON.stringify(value)}`;
		else return `${stringKey}=${encodeURIComponent(value)}`;
	});

	if (queryStrings.length === 0) return '';
	return '?' + queryStrings.join('&');
};
