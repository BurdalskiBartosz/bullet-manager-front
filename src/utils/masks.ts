export type tMask = (string | RegExp)[];

export const dateMask: tMask = [
	/\d/,
	/\d/,
	'/',
	/\d/,
	/\d/,
	'/',
	/\d/,
	/\d/,
	/\d/,
	/\d/
];
