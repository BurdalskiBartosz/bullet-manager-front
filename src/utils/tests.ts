import { render } from '@testing-library/react';
import Providers from '../providers';

const customRender = (ui: any) => render(ui, { wrapper: Providers });

export * from '@testing-library/react';

export { customRender as render };
