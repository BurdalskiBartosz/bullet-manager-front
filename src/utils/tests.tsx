import { render, RenderOptions } from '@testing-library/react';
import { PropsWithChildren } from 'react';
import { Provider } from 'react-redux';
import { PreloadedState } from '@reduxjs/toolkit';
import { AppStore, RootState, setupStore } from 'store/store';
import { ThemeProvider } from 'styled-components';
import { theme } from 'styles/theme';
import { BrowserRouter } from 'react-router-dom';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
	preloadedState?: PreloadedState<RootState>;
	store?: AppStore;
}
const reduxStore = setupStore();
export const renderWithProviders = (
	ui: React.ReactElement,
	{
		preloadedState = {},
		// Automatically create a store instance if no store was passed in
		store = reduxStore,
		...renderOptions
	}: ExtendedRenderOptions = {}
) => {
	const Wrapper = ({ children }: PropsWithChildren<{}>) => {
		return (
			<BrowserRouter>
				<Provider store={store}>
					<ThemeProvider theme={theme}>{children}</ThemeProvider>
				</Provider>
			</BrowserRouter>
		);
	};

	// Return an object with the store and all of RTL's query functions
	return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export { renderWithProviders as render };
export * from '@testing-library/react';
