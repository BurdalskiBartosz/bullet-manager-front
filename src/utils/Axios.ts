/* eslint-disable no-param-reassign */
import axios, { AxiosRequestConfig, AxiosError, AxiosResponse } from 'axios';
import { baseURL } from './constants';

const options: AxiosRequestConfig = {
	baseURL,
	timeout: 10000,
	headers: { 'Content-Type': 'application/json' },
	withCredentials: true
};

export const fetchProvider = axios.create(options);
const { interceptors } = fetchProvider;

const getRequestConfig = (config: AxiosRequestConfig) => {
	const token: string | null = window.localStorage.getItem('token');

	if (token) config.headers!.Authorization = `Bearer ${token}`;

	return config;
};

const getRequestError = (error: AxiosError) => Promise.reject(error);

const getResponse = (response: AxiosResponse) => response;

const getResponseError = (error: AxiosError) => {
	const errorResponse =
		error.response?.status === 400 ? error : error.response;
	return Promise.reject(errorResponse);
};

interceptors.request.use(getRequestConfig, getRequestError);
interceptors.response.use(getResponse, getResponseError);

export const instanceForMocks = axios.create(options);
