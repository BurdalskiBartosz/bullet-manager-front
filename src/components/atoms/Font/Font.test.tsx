import { render, screen } from 'utils/tests';
import Font from './Font';

const styleProp = {
	color: '#DDDDDD',
	fontSize: '22px',
	marginBottom: '20px'
};

describe('Font', () => {
	it('should render correctly without any props', () => {
		render(<Font>Font</Font>);
		const fontComponent = screen.getByText(/Font/i);
		expect(fontComponent).toMatchSnapshot();
	});

	it('should render correctly with style prop', () => {
		render(<Font style={styleProp}>Font</Font>);
		const fontComponent = screen.getByText(/Font/i);
		expect(fontComponent).toHaveStyleRule('color', '#DDDDDD');
		expect(fontComponent).toHaveStyleRule('font-size', '22px');
		expect(fontComponent).toHaveStyleRule('margin-bottom', '20px');
	});

	it('should correctly load styles from variant', () => {
		render(<Font variant="label">Font</Font>);
		const fontComponent = screen.getByText(/Font/i);
		expect(fontComponent).toHaveStyleRule('font-size', '1.2rem');
		expect(fontComponent).toHaveStyleRule('line-height', '1.6rem');
	});

	it('should correctly load styles from variant with breakpoints', () => {
		render(<Font variant="header">Font</Font>);
		const fontComponent = screen.getByText(/Font/i);
		expect(fontComponent).toHaveStyleRule('font-size', '3.5rem');
		expect(fontComponent).toHaveStyleRule('font-size', '4rem', {
			media: '(min-width: 576px)'
		});
		expect(fontComponent).toHaveStyleRule('font-size', '5rem', {
			media: '(min-width: 992px)'
		});
	});

	it('should correctly override variant styles with style prop', () => {
		render(
			<Font variant="header" style={styleProp}>
				Font
			</Font>
		);
		const fontComponent = screen.getByText(/Font/i);
		expect(fontComponent).toHaveStyleRule('font-size', '22px');
		expect(fontComponent).toHaveStyleRule('margin-bottom', '20px');
	});
});
