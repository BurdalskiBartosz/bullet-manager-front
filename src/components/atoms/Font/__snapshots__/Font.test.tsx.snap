// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Font should render correctly without any props 1`] = `
<p
  class=""
>
  Font
</p>
`;
